### ISSUE TITLE
[CRYO-49] Develop new code tutorial demonstrating access and visualization of MEaSUREs glacier data

### ISSUE THREAD

**Issue by IanSinColorado** [nsidc/JIRA-GitHub-ETL#18]
JIRA Issue: https://nsidc.org/jira/browse/CRYO-49
Original Reporter: asteiker (JIRA username)
 *** 
<p><a href="https://trello.com/c/qaVpOq5b/38-develop-new-code-tutorial-demonstrating-access-and-visualization-of-measures-glacier-data" class="external-link" target="_blank" rel="nofollow noopener">https://trello.com/c/qaVpOq5b/38-develop-new-code-tutorial-demonstrating-access-and-visualization-of-measures-glacier-data</a></p>

<p>Acceptance Criteria:</p>
<ul>
	<li>Measures jupyter notebook tutorial is added to <a href="https://github.com/nsidc/NSIDC-Data-Tutorials/tree/main/notebooks" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/nsidc/NSIDC-Data-Tutorials/tree/main/notebooks</a> </li>
	<li>Tutorial runs successfully in all environment options specified in the Readme (Binder, Conda, Docker)</li>
	<li>Any other TBD Acceptance Criteria for new code-based tutorials as created under "Best practices guide and template for interactive code tutorials"</li>
</ul>

<p> </p>

## Checklist:
 - [ ] Check/modify tutorial so it works in NSIDC tutorial environment
 - [ ] Update tutorial so it matches the tutorial template
 - [ ] Put tutorial through review process (would like to include a review from the PI for the data sets)
 - [ ] Move final version to repo in GitHub

### JIRAUSER20800 (JIRA username) added a comment on Thu, 24 Feb 2022 07:59:11
 <p><a href="https://nsidc.org/jira/secure/ViewProfile.jspa?name=asteiker" class="user-hover" rel="asteiker">Amy Steiker</a> I was testing this tutorial in the NSIDC tutorials environment and have run into an issue. I use the module 'pycrs' and this is not included in the NSIDC tutorials environment. As I see it there are 2 options, add this module to the tutorials environment, or I can try and find a workaround for the step that uses this module? Or maybe there are other options?</p>

<p>Also, it might be good to address this potential stumbling block in the guide, as in we recommend that before starting to develop the tutorial it is a good idea to review the libraries that are in already in the tutorials environment, should they try use only those libraries? And what to do if the tutorial requires a library that is not in the tutorials environment.</p>

 ### asteiker (JIRA username) added a comment on Wed, 2 Mar 2022 11:00:58
 <p>What is pycrs doing in the notebook?</p>
<ul>
	<li>Used to write the correct crs metadata to a newly resampled and subset geotiff but Jennie needs to double-check code</li>
	<li><a href="https://nsidc.org/jira/secure/ViewProfile.jspa?name=apbarret" class="user-hover" rel="apbarret">Andy Barrett</a> will review notebook to provide guidance on whether to move forward with pycrs or if there's an easier/better approach.</li>
	<li>To test pycrs:
	<ul>
		<li>Clone environment</li>
		<li>Add pycrs </li>
		<li>Test locally to ensure no conflicts with existing packages</li>
	</ul>
	</li>
</ul>

 ### JIRAUSER20800 (JIRA username) added a comment on Thu, 10 Mar 2022 13:14:41
 <p>Currently blocked due to an error running it in the tutorials environment. See attached screenshot for error message. Have tried to find a work around but it is not working so I have reached out to Luis to see if he has come across this error before.  <span class="image-wrap" style=""><img src="https://nsidc.org/jira/secure/attachment/131202/131202_Screen+Shot+2022-03-10+at+1.08.13+PM.png" style="border: 0px solid black" /></span></p>

 ### asteiker (JIRA username) added a comment on Wed, 23 Mar 2022 10:53:18
 <p>Status 3/23:</p>
<ul>
	<li>Still blocked when attempting to add `earthdata` to data-tutorials environment.
	<ul>
		<li>Luis had also worked to resolve this for IS2 Hackweek environment, so maybe we could leverage that fix.</li>
	</ul>
	</li>
	<li>May make more sense to update the tutorial to utilize `earthdata` instead. 
	<ul>
		<li>Should hold off on those updates until <a href="https://nsidc.org/jira/browse/CRYO-52" title="As a tutorial creator, I want to utilize a template so that my tutorial follows a standardized structure" class="issue-link" data-issue-key="CRYO-52"><del>CRYO-52</del></a>, <a href="https://nsidc.org/jira/browse/CRYO-54" title="As a tutorial creator, I want to know the steps needed to publish a new tutorial on one of NSIDC&#39;s code tutorial github repo&#39;s" class="issue-link" data-issue-key="CRYO-54"><del>CRYO-54</del></a>, and <a href="https://nsidc.org/jira/browse/CRYO-53" title="As a tutorial creator, I want to know how to contribute to our code tutorial github repositories" class="issue-link" data-issue-key="CRYO-53"><del>CRYO-53</del></a> are in place so that this can be truly used as a test case for our best practices guide</li>
		<li>Support for `earthdata` is also something we should consider here, as currently the library is considered "low level of support" on the github repo.</li>
	</ul>
	</li>
</ul>

 ### JIRAUSER20800 (JIRA username) added a comment on Tue, 29 Mar 2022 16:44:04
 <p>Another new blocker to this is that one of the data sets I use is versioning up and the changes to the new version will affect the notebook. So I will have to wait for this new version to be finalized and published.</p>

 ### asteiker (JIRA username) added a comment on Wed, 13 Apr 2022 09:37:01
 <p><a href="https://nsidc.org/jira/secure/ViewProfile.jspa?name=lopez" class="user-hover" rel="lopez">Luis Lopez</a> will update the environment .yml in Main, and then <a href="https://nsidc.org/jira/secure/ViewProfile.jspa?name=jero7025" class="user-hover" rel="jero7025">Jennifer Roebuck</a> can update her measures branch with that updated environment. </p>

 ### asteiker (JIRA username) added a comment on Wed, 18 May 2022 10:56:24
 <p>Standup update 5/18:</p>
<ul>
	<li>Remaining blocked until this can be re-written with the best practices components ( template and contributor guide)</li>
	<li><a href="https://nsidc.org/jira/secure/ViewProfile.jspa?name=jero7025" class="user-hover" rel="jero7025">Jennifer Roebuck</a> will push up comments on what needs to be re-written to the measures branch</li>
</ul>

<p>Questions:</p>
<ul>
	<li>Do we have instructions for how to copy the template to create a new notebook?
	<ul>
		<li>There is an ability through Github to create a template for a new fork (Scott Henderson / eSciences is utilizing this) &lt;- We should write a new ticket in our backlog to pursue this</li>
		<li>Currently only roughly outlined in the Best Practices Guide </li>
	</ul>
	</li>
</ul>

 ### JIRAUSER20800 (JIRA username) added a comment on Fri, 20 May 2022 15:30:01
 <p><a href="https://nsidc.org/jira/secure/ViewProfile.jspa?name=asteiker" class="user-hover" rel="asteiker">Amy Steiker</a> have pushed the notebook with comments on what needs to change to the measures branch. I also met with Danica to go over the changes and the notebook in general. I said the plan was to use this as a test case for the new best practices guide and that going forward it would be best to rewrite the notebook in the new template once it is finalized and available. I mentioned you may reach out about it whilst I am on leave.</p>

 ### asteiker (JIRA username) added a comment on Wed, 25 May 2022 10:29:11
 <p>Thank you, <a href="https://nsidc.org/jira/secure/ViewProfile.jspa?name=jero7025" class="user-hover" rel="jero7025">Jennifer Roebuck</a> ! I am going to move this ticket out of our Epic given that you completed all the work you're able to do without the Best Practices Guide complete yet. Adding the measures branch here for reference:  <a href="https://github.com/nsidc/NSIDC-Data-Tutorials/tree/measures" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/nsidc/NSIDC-Data-Tutorials/tree/measures</a> .</p>

 ### asteiker (JIRA username) added a comment on Wed, 25 May 2022 11:10:39
 <p>Moving back to backlog. Needs coordination with Measures team and Danica to revisit this once the Best Practices Guide is complete.</p>

 ### JIRAUSER20800 (JIRA username) added a comment on Wed, 5 Apr 2023 13:56:08
 <p>Have asked Danica to review</p>

 ### asteiker (JIRA username) added a comment on Tue, 9 May 2023 16:59:30
 <p>Testing was successful using the Openscapes Hub, manually importing the latest earthaccess version:</p>

<p>%pip install git+<a href="https://github.com/nsidc/earthaccess.git@auth-improvements" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/nsidc/earthaccess.git@auth-improvements</a></p>

<p>Unfortunately I am getting an error when attempting to run the Binder build from the PR. Note that <a href="https://nsidc.org/jira/browse/CRYO-64" title="As a user I want to know if any of the notebooks run without errors" class="issue-link" data-issue-key="CRYO-64">CRYO-64</a>/67 were merged into the measures branch, so there were also failures in the automated notebook runs, however I think this is an issue with the tests themselves, since I confirmed that the notebook is running end to end using the correct earthaccess version. I attempted to test the branch using Docker but I also ran into an error (no module found when I imported earthaccess). I will need Luis's guidance to understand why I'm getting the discrepancy in the quarto tests.</p>

<p> </p>
